### PR TITLE
python3Packages.coapthon3: init at 1.0.1

### PR DESCRIPTION
**Pull request by urbas**
###### Motivation for this change

The `coapthon3` Python package is not yet packaged.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after) init at 452.2K
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -9041,6 +9041,12 @@
     githubId = 619015;
     name = "Svintsov Dmitry";
   };
+  urbas = {
+    email = "matej.urbas@gmail.com";
+    github = "urbas";
+    githubId = 771193;
+    name = "Matej Urbas";
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:9048]: Please follow the contributions guide and put this into a separate commit.

File: pkgs/development/python-modules/coapthon3/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{ fetchFromGitHub
+, lib
+, pythonPackages }:
+
+pythonPackages.buildPythonPackage (with pythonPackages; rec {
+  pname = "CoAPthon3";
+  version = "1.0.1";
+
+  propagatedBuildInputs = [ cachetools ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/coapthon3/default.nix:9]: This belongs after fetchPypi.

File: pkgs/development/python-modules/coapthon3/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1w6bwwd3qjp4b4fscagqg9wqxpdgvf4sxgzbk2d2rjqwlkyr1lnx";
+  };
+
+  meta = with lib; {
+    description = "CoAPthon3 is a python3 library to the CoAP protocol compliant with the RFC.";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/coapthon3/default.nix:17]:
```suggestion
    description = "Python3 library to the CoAP protocol compliant with the RFC";
```

File: pkgs/development/python-modules/coapthon3/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    homepage    = "https://github.com/Tanganelli/${pname}";
+    license     = licenses.mit;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/coapthon3/default.nix:19]:
```suggestion
    homepage = "https://github.com/Tanganelli/${pname}";
    license = licenses.mit;
```

File: pkgs/development/python-modules/coapthon3/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{ fetchFromGitHub
+, lib
+, pythonPackages }:
```
Comment (urbas) [inline pkgs/development/python-modules/coapthon3/default.nix:3]: `fetchFromGitHub` is unused. Will remove.
```suggestion
{ lib
, pythonPackages }:
```

File: pkgs/development/python-modules/coapthon3/default.nix [hunk 6]
```diff
@@ -0,0 +1,20 @@
+{ lib, pythonPackages }:
+
+pythonPackages.buildPythonPackage (with pythonPackages; rec {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/coapthon3/default.nix:3]:
```suggestion
{ lib, pythonPackages, cachetools }:

pythonPackages.buildPythonPackage rec {
```

This is usually not done because I think it breaks the compatibility with other python versions. The correct way is to pass in the other python modules directly.

Comment (urbas) [reply pkgs/development/python-modules/coapthon3/default.nix]: Done. I did the same with `fetchPypi` and `buildPythonPackages`. Also removed the `)`.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104528` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python27Packages.coapthon3</li>
    <li>python37Packages.coapthon3</li>
    <li>python38Packages.coapthon3</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104528` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python27Packages.coapthon3</li>
    <li>python37Packages.coapthon3</li>
    <li>python38Packages.coapthon3</li>
  </ul>
</details>
